### PR TITLE
Include fields for operating system and ami id in environment create on API

### PR DESCRIPTION
**Pull request by zpatrick**
closes https://github.com/quintilesims/layer0/issues/152
(partially closes) https://github.com/quintilesims/layer0/issues/158

### CODE REVIEW

File: api/logic/environment_logic.go [hunk 1]
```diff
@@ -85,7 +86,17 @@ func (e *L0EnvironmentLogic) CreateEnvironment(req models.CreateEnvironmentReque
...
-	environment, err := e.Backend.CreateEnvironment(req.EnvironmentName, req.InstanceSize, req.MinClusterCount, req.UserDataTemplate)
+	if req.OperatingSystem == "" {
+		return nil, errors.Newf(errors.MissingParameter, "OperatingSystem is required")
+	}
+
+	environment, err := e.Backend.CreateEnvironment(
+		req.EnvironmentName,
+		req.InstanceSize,
+		req.OperatingSystem,
+		req.AMIID,
```
Comment (diemonster) [inline api/logic/environment_logic.go]: [nit] I guess the guideline here (https://msdn.microsoft.com/en-us/library/141e06ef(v=vs.71).aspx) would be to use `AmiID`. although that looks even weirder

File: common/config/config.go [hunk 2]
```diff
@@ -123,22 +126,16 @@ func SetCLIVersion(version string) {
 	cliVersion = version
 }
 
-var serviceAMIs = map[string]string{
-	"us-west-2": "ami-6cb9ac0d",
```
Comment (diemonster) [inline common/config/config.go]: are we no longer hardcoding default AMIs?

File: api/backend/ecs/environment_manager.go [hunk 3]
```diff
@@ -374,7 +397,75 @@ func renderUserData(ecsEnvironmentID id.ECSEnvironmentID, userData []byte) (stri
...
+Read-S3Object -BucketName {{ .S3Bucket }} -Key bootstrap/dockercfg -File dockercfg.json
+$dockercfgContent = [IO.File]::ReadAllText("dockercfg.json")
+[Environment]::SetEnvironmentVariable("ECS_ENGINE_AUTH_DATA", $dockercfgContent, "Machine")
+[Environment]::SetEnvironmentVariable("ECS_ENGINE_AUTH_TYPE", "dockercfg", "Machine")
+
+# Output all environment variables
+Get-ChildItem -Path Env:* | Sort-Object Name
+
+# Output all environment variables
+Get-ChildItem -Path Env:* | Sort-Object Name
```
Comment (zpatrick) [inline api/backend/ecs/environment_manager.go]: fixed

File: common/config/config.go (hunk 2 above)
Comment (zpatrick) [inline common/config/config.go]: this was actually unused. it duplicated what was done in l0-setup

**Pull request merged**